src/actions/posts.js

```diff
@@ -364,6 +364,40 @@ export function editPost(post) {
     });
 }
 
+export function setUnreadPost(userId, postId) {
+    return async (dispatch, getState) => {
+        dispatch({type: ChannelTypes.POST_UNREAD_REQUEST}, getState);
+
+        let unreadChan;
+        const state = getState();
+        try {
+            unreadChan = await Client4.markPostAsUnread(userId, postId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
+                {type: PostTypes.POST_UNREAD_FAILURE, error},
+                logError(error),
+            ]), getState);
+            return {error};
+        }
+
+        const oldMsgCount = state.entities.channels.myMembers[unreadChan.channel_id].msg_count;
```

Comment:
Is there any chance that `state.entities.channels.myMembers[unreadChan.channel_id]` might end up being undefined? I'm wondering if this could use a null check at all

Reply:
It shouldn't happen as we are considering an action which happens after viewing the channel, but that check won't hurt. So again I'll err on the safe side and add the check :)